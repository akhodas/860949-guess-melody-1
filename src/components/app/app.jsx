import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from "react-redux";

const Type = {
  ARTIST: `game--artist`,
  GENRE: `game--genre`,
};


class App extends Component {
  render() {
    const {
      questions,
      renderScreen,
      step,
    } = this.props;

    return (
      <section className={`game ${Type.ARTIST}`}>
        {renderScreen(questions[step])}
      </section>
    );
  }
}

App.propTypes = {
  gameTime: PropTypes.number.isRequired,
  questions: PropTypes.array.isRequired,
  renderScreen: PropTypes.func.isRequired,
  step: PropTypes.number.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  questions: state.questions,
  step: state.step,
});


export {App};

export default connect(mapStateToProps)(App);
