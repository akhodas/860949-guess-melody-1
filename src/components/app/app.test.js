import React from 'react';
import renderer from 'react-test-renderer';
import {App} from './app.jsx';

describe(`App`, () => {
  it(`correctly rendered`, () => {
    const tree = renderer
      .create(<App
        errorCount={0}
        gameTime={0}
        startGame={jest.fn()}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

