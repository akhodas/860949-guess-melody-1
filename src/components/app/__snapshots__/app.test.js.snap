// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App correctly renders artist question screen 1`] = `
<section
  className="game game--artist"
/>
`;

exports[`App correctly renders first screen 1`] = `
<section
  className="game game--artist"
/>
`;

exports[`App correctly renders genre question screen 1`] = `
<section
  className="game game--artist"
/>
`;
